src/bson.ts

```diff
@@ -86,7 +85,7 @@ export function pluckBSONSerializeOptions(options: BSONSerializeOptions): BSONSe
  */
 export function resolveBSONOptions(
   options?: BSONSerializeOptions,
-  parent?: OperationParent
+  parent?: { bsonOptions?: BSONSerializeOptions }
```

Comment:
This removes a circular dependency wrt TypeScript types only, since everything imports BSON, everything was importing the `OperationParent`.